cpp/ql/test/library-tests/dataflow/taint-tests/vector.cpp

```diff
@@ -141,3 +141,30 @@ void test_vector_clear() {
 	sink(v3); // [FALSE POSITIVE]
 	sink(v4);
 }
+
+void taint_vector_output_iterator(std::vector<int>::iterator iter) {
+	*iter = source();
+}
+
+void test_vector_output_iterator() {
+	std::vector<int> v1(10), v2(10), v3(10), v4(10);
+
+	std::vector<int>::iterator i1 = v1.begin();
+	*i1 = source();
+	sink(v1); // tainted [NOT DETECTED by IR]
+
+	for(std::vector<int>::iterator it = v2.begin(); it != v2.end(); ++it) {
+		*it = source();
+	}
+	sink(v2); // tainted [NOT DETECTED by IR]
+
+	for(int& x : v3) {
+		x = source();
+	}
+	sink(v3); // tainted [NOT DETECTED]
+
+	for(std::vector<int>::iterator it = v4.begin(); it != v4.end(); ++it) {
+		*it = source();
+	}
+	sink(v4); // tainted [NOT DETECTED by IR]
```

Comment:
I can't see any difference between the test with `v4 and the test with `v2`.

Apart from that these are great tests and really show the value of this PR.

Reply:
I see that you've fixed this now - thanks!